src/portable/portable.c

```diff
@@ -420,7 +449,16 @@ static int portable_extract_by_path(
                                 return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Invalid item sent from child.");
 
-                        add = portable_metadata_new(iov_buffer, path, fd);
+                        /* Given recvmsg cannot be used with multiple io vectors if you don't know the size in advance,
+                         * use a marker to separate the name and the optional SELinux context. */
+                        char *selinux_label = strchr(iov_buffer, '\0');
+                        assert(selinux_label);
+                        if (n > selinux_label - iov_buffer + 1)
```

Comment:
given the buffer is NUL terminated this check is kinda unnecessary, no? would suffix to nullify selinux_label if isempty() holds, no?